rust/src/lib.rs

```diff
@@ -70,6 +70,13 @@ fn dir_from_dfd(fd: libc::c_int) -> io::Result<openat::Dir> {
     Ok(r)
 }
 
+// It's not really &'static of course...but we can't
+// tell Rust about our lifetimes from the C side.
+fn tf_from_raw(tf: *mut Treefile) -> &'static mut Treefile {
```

Comment:
Isn't there a way to make this have the same lifetime as the `tf` parameter that was passed? Not that it makes a difference given that we'd only be calling this in places where `tf` is alread a parameter at the callsite. Though it seems nicer than `&'static`.

Reply:
Raw pointers [don't have lifetimes](https://doc.rust-lang.org/book/first-edition/raw-pointers.html).

> lack any form of lifetimes, unlike &, and so the compiler cannot reason about dangling pointers; and